libminifi/include/core/TypedValues.h

```diff
@@ -141,6 +141,47 @@ class TimePeriodValue : public TransformableValue, public state::response::UInt6
  * format <numeric> <byte size>.
  */
 class DataSizeValue : public TransformableValue, public state::response::UInt64Value {
+private:
+
+  //Signed
+  template<typename T, typename std::enable_if<std::is_signed<T>::value, T>::type* = nullptr>
+  static bool AssignIfFits(int64_t base, uint64_t mult, T& output) {
+    int64_t val = base * mult;
+    if(val >= std::numeric_limits<T>::min() && val <= std::numeric_limits<T>::max()) {
+      output = val;
+      return true;
+    }
+    return false;
+  }
+
+  //Unsigned
+  template<typename T, typename std::enable_if<!std::is_signed<T>::value, T>::type* = nullptr>
+  static bool AssignIfFits(uint64_t base, uint64_t mult, T& output) {
+    uint64_t val = base * mult;
+    if(val <= std::numeric_limits<T>::max()) {
+      output = val;
+      return true;
+    }
+    return false;
+  }
+
+  //Signed
+  template<typename T, typename std::enable_if<std::is_signed<T>::value, T>::type* = nullptr>
+  static bool StringToNum(const char* str_begin, char **str_end, T& output) {
+    int64_t val = std::strtoll(str_begin, str_end, 0);
+    return AssignIfFits<T>(val, 1, output);
+  }
+
+  //Unsigned
+  template<typename T, typename std::enable_if<!std::is_signed<T>::value, T>::type* = nullptr>
+  static bool StringToNum(const char* str_begin, char **str_end, T& output) {
+    if(str_begin[0] == '-') {
```

Comment:
can't assume that it's zero, and that it's non-empty

Reply:
Good shout, thanks!
This also means adding a tectcase to check with empty and invalid strings to make sure it handles errors gracefully.